### PR TITLE
Document the new unarchive pipeline REST API endpoint

### PR DESCRIPTION
**Pull request by shevaun**
Implementation here: https://github.com/buildkite/buildkite/pull/6627

### CODE REVIEW

File: pages/apis/rest_api/pipelines.md.erb [hunk 1]
```diff
@@ -820,6 +820,88 @@ Error responses:
 </tbody>
 </table>
 
+## Unarchive a pipeline
+
+Restore an archived pipeline.
```
Comment (plaindocs) [inline pages/apis/rest_api/pipelines.md.erb:825]: Unless restore has more meaning than unarchive, lets stick to the same wording so folks don't worry if there is a difference. And it maybe makes sense to reword the description as well

> Unarchived pipeles are editable, and are shown on the Pipeline pages.

Comment (plaindocs) [reply pages/apis/rest_api/pipelines.md.erb:825]: Otherwise, great stuff!

**Comment by shevaun** [comment]
@plaindocs not sure what to do with this - the linting check does not like `unarchive` apparently 😅

**Comment by JuanitoFatas** [comment]
> @plaindocs not sure what to do with this - the linting check does not like `unarchive` apparently 😅

Add `unarchived` after https://github.com/buildkite/docs/blob/f09b3e5ded6a9b1e403e0e78469e2c291846c041/vale/vocab.txt#L123 should work

**Review by plaindocs — APPROVED** [review]
Awesome! Thanks for kicking the linter @JuanitoFatas !

**Pull request merged**